### ISSUE TITLE
Sidebar menu with dynamic content does not show default tab

### ISSUE THREAD

**Issue by maelick** [rstudio/shinydashboard#71]
When using a sidebar menu dynamically generated no tab is shown by default.

Here's the example from the [doc](https://rstudio.github.io/shinydashboard/structure.html#dynamic-content-1) with some text added inside the dashboard bod to be able to replicate the issue:

``` R
library(shiny)
library(shinydashboard)

ui <- dashboardPage(
  dashboardHeader(title = "Dynamic sidebar"),
  dashboardSidebar(
    sidebarMenuOutput("menu")
  ),
  dashboardBody(tabItems(
    tabItem(tabName = "dashboard", h2("Dashboard tab content"))
  ))
)

server <- function(input, output) {
  output$menu <- renderMenu({
    sidebarMenu(id="mytabs",
      menuItem("Menu item", tabName="dashboard", icon = icon("calendar"))
    )
  })
}

shinyApp(ui, server)
```

By default it does not show the tab item "dashboard" and the user has to clic on it to show it. I tried to manually fix this in my app using updateTabItems but it does not work either because the sidebar menu object is not stored in the input variable (but it was in the static version).

I tested this both with the CRAN and last Github versions of shiny and shinydashboard.

**Comment by tfrojd** [comment]
Same issue here.

**Comment by ChanningWong** [comment]
The same problem occured

**Comment by danwwilson** [comment]
It would be great to get a solution to this.

**Comment by funnng** [comment]
Same issue here. when this issue will be fixed?

**Comment by isaacfab** [comment]
This is still not fixed.. but here is a simple work around if your default page is static or can be pre-built without reactivity.  You can list multiple sidebarMenu Items in a row.  
`library(shiny)
library(shinydashboard)

ui <- dashboardPage(
  dashboardHeader(title = "Dynamic sidebar"),
  dashboardSidebar(
    sidebarMenu(
      menuItem("Default Menu item", tabName="intro_page", icon = icon("info"))
    ),
    sidebarMenuOutput("menu")
  ),
  dashboardBody(tabItems(
      tabItem(tabName = "intro_page",
              fluidRow(box(helpText("Here is your default page!"),width=12))
      ),
      tabItem(tabName = "dashboard", h2("Dashboard tab content"))
  )
  )
)

server <- function(input, output) {
  output$menu <- renderMenu({
    sidebarMenu(id="mytabs",
                menuItem("Menu item", tabName="dashboard", icon = icon("calendar"))
    )
  })
}

shinyApp(ui, server)`

**Comment by sbstern** [comment]
Bump on this. It makes shinydashboard almost unusable for what I am trying to do...

**Comment by funnng** [comment]
i used shinyjs for a temporary fix
`shinyjs.initPage = function(){setTimeout(function(){$("#sidebarmenu > li:nth-child(1) > a").click();}, 1500);}`

**Comment by mayeromain** [comment]
Here is a solution using `updateTabItems`.
You can also find it on stackoverflow.

```
library(shiny)
library(shinydashboard)
ui <- dashboardPage(
  dashboardHeader(title = "Dynamic sidebar"),
  dashboardSidebar(
    sidebarMenu(id="tabs",
    sidebarMenuOutput("menu")
    )
  ),
  dashboardBody(
    tabItems(
      tabItem(tabName = "m1", p("Menu content 1") ),
      tabItem(tabName = "m2", p("Menu content 2") )
    )
  )
)
server <- function(input, output,session) {

  output$menu <- renderMenu({
    sidebarMenu(
           menuItem("Menu item1", tabName="m1", icon = icon("calendar")),
           menuItem("Menu item2", tabName="m2", icon = icon("database"))
           )
  })
  isolate({updateTabItems(session, "tabs", "m2")})
}
shinyApp(ui, server)
```

 http://stackoverflow.com/questions/36742932/r-shinydashboard-dynamic-menu-selection/36913826#36913826

**Comment by bborgesr** [comment]
Fixed in #199
